ot/mpz-ot/src/actor/kos/receiver.rs

```diff
@@ -0,0 +1,346 @@
+use std::collections::{HashMap, VecDeque};
+
+use async_trait::async_trait;
+use futures::{
+    channel::{mpsc, oneshot},
+    stream::Fuse,
+    SinkExt, StreamExt,
+};
+use utils_aio::{
+    non_blocking_backend::{Backend, NonBlockingBackend},
+    sink::IoSink,
+    stream::IoStream,
+};
+
+use crate::{
+    kos::{Receiver, ReceiverError, ReceiverKeys},
+    OTError, OTReceiverShared, OTSetup, VerifiableOTReceiverShared, VerifiableOTSender,
+};
+use mpz_core::{Block, ProtocolMessage};
+use mpz_ot_core::kos::{msgs::SenderPayload, PayloadRecord};
+
+use crate::actor::kos::{
+    into_kos_sink, into_kos_stream,
+    msgs::{ActorMessage, Message, TransferPayload, TransferRequest},
+    ReceiverActorError,
+};
+
+enum Command {
+    Receive {
+        id: String,
+        choices: Vec<bool>,
+        caller_response: oneshot::Sender<Result<(ReceiverKeys, SenderPayload), ReceiverError>>,
+    },
+    Verify(Verify),
+    Shutdown(Shutdown),
+}
+
+struct Verify {
+    id: String,
+    caller_response: oneshot::Sender<Result<PayloadRecord, ReceiverError>>,
+}
+
+struct PendingTransfer {
+    keys: ReceiverKeys,
+    caller_response: oneshot::Sender<Result<(ReceiverKeys, SenderPayload), ReceiverError>>,
+}
+
+opaque_debug::implement!(PendingTransfer);
+
+struct Shutdown {
+    caller_response: oneshot::Sender<Result<(), ReceiverActorError>>,
+}
+
+#[derive(Default)]
+struct State {
+    ids: HashMap<String, u32>,
+
+    pending_transfers: HashMap<String, PendingTransfer>,
+    pending_verify: VecDeque<Verify>,
+}
+
+/// KOS receiver actor.
+pub struct ReceiverActor<BaseOT, Si, St> {
+    sink: Si,
+    stream: Fuse<St>,
+
+    receiver: Receiver<BaseOT>,
+
+    state: State,
+
```

Comment:
/// The sending end of the channel used by the controller to send commands to this actor.